### PR TITLE
Fix: wrap postMessage in timeout to prevent race conditions

### PR DESCRIPTION
**Pull request by AustP**
### Description

There was a race condition where sometimes Kibisis wouldn't show up in the available providers. I believe this was happening because the channel was being created and a listener was being attached and then a message was immediately being posted. I've made a change so that the message being posted happens on the next tick so it actually gives the channel time to setup.

@kieranroneill 

### Checklist

- [x] Code compiles correctly
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by drichar — APPROVED** [review]
I have noticed this happening occasionally, and the randomness of it does suggest a race condition in the initialization logic. Have you seen this before @kieranroneill ?

The issue is sporadic to begin with... that said, I can't reproduce it with your solution. So it does seem to do the trick 👍 

Thank you for the PR! Sorry about the delay reviewing this and https://github.com/TxnLab/use-wallet/pull/141

**Comment by drichar** [comment]
This is approved but I'd like to see if @kieranroneill has any feedback about the issue before merging (ty in advance!)

**Comment by kieranroneill** [comment]
@drichar I will have a look at this today 😊

**Pull request merged**
